Warp.WebApp/Pages/Index.cshtml.cs

```diff
@@ -19,22 +24,67 @@ public IndexModel(ILoggerFactory loggerFactory, IEntryService entryService) : ba
 
 
     [OutputCache(Duration = 3600)]
-    public IActionResult OnGet()
+    public async Task<IActionResult> OnGet(string? id, CancellationToken cancellationToken)
     {
-        OpenGraphModel = OpenGraphService.GetDefaultModel();
+        if (!string.IsNullOrEmpty(id))
+        {
+            var decodedId = IdCoder.Decode(id);
+            if (decodedId == Guid.Empty)
+                return RedirectToError(ProblemDetailsHelper.Create("Can't decode a provided ID."));
+
+            if (this.HttpContext.User.Claims.FirstOrDefault(x => x.Type == ClaimTypes.UserData && x.Value == decodedId.ToString()) != null)
+            {
+                var (_, isFailure, entry, problemDetails) = await _entryService.Get(decodedId, cancellationToken);
+                if (isFailure)
+                    return RedirectToError(problemDetails);
+                BuildModel(id, entry);
+                AddOpenGraphModel();
+                return Page();
+            }
+        }
 
+        OpenGraphModel = OpenGraphService.GetDefaultModel();
         return Page();
+
+
+        void BuildModel(string entryId, EntryInfo entryInfo)
+        {
+            TextContent = entryInfo.Entry.Content;
+            ImageIds = entryInfo.ImageIds;
+        }
+
+        void AddOpenGraphModel()
+        {
+            OpenGraphModel = OpenGraphService.GetModel(TextContent);//, ImageUrls);
+        }
     }
 
 
     public async Task<IActionResult> OnPost(CancellationToken cancellationToken)
     {
         var expiresIn = GetExpirationPeriod(SelectedExpirationPeriod);
         var (_, isFailure, id, problemDetails) = await _entryService.Add(TextContent, expiresIn, ImageIds, cancellationToken);
-        
-        return isFailure 
-            ? RedirectToError(problemDetails) 
-            : RedirectToPage("./Entry", new { id = IdCoder.Encode(id) });
+
+        if (isFailure)
+            return RedirectToError(problemDetails);
```

Comment:
Omit braces only when both parts of an expression are one-liners, otherwise use them everywhere